services/api/src/resources/project/resolvers.js

```diff
@@ -193,6 +193,68 @@ const getProjectByName = async (
   }
 };
 
+
+const getProjectsByMetadata = async (
+  root,
+  args,
+  {
+    sqlClient,
+    hasPermission,
+  },
+) => {
+
+  let where;
+  try {
+    await hasPermission('project', 'viewAll');
+
+    where = whereAnd([
+      args.createdAfter ? 'created >= :created_after' : '',
+      args.gitUrl ? 'git_url = :git_url' : '',
+    ]);
+  } catch (err) {
+    if (!keycloakGrant) {
+      logger.warn('No grant available for getAllProjects');
+      return [];
+    }
+
+    const userProjectIds = await models.UserModel.getAllProjectsIdsForUser({
+      id: keycloakGrant.access_token.content.sub,
+    });
+
+    where = whereAnd([
+      args.createdAfter ? 'created >= :created_after' : '',
+      args.gitUrl ? 'git_url = :git_url' : '',
+      inClause('id', userProjectIds),
+    ]);
+  }
+
+  for (const meta of args['metadata']) {
+
+    let key = meta['key'];
+    let val = meta['value'];
+
+    var q = `JSON_CONTAINS(metadata, '"`+val+`"', '$.`+key+`')`;
```

Comment:
wow, never heard of `JSON_CONTAINS` in MySQL? Crazy world :D

Reply:
ha yeah, sort-of relational/nosql hybrid